### ISSUE TITLE
PopoverContent doesn't receive focus when using Popover with trigger prop set to 'hover' [a11y]

### ISSUE THREAD

**Issue by adamraya** [chakra-ui/chakra-ui#3820]
# 🐛 Bug report
Using the `Popover` component with the prop `trigger={'hover'}` doesn't send the focus to the `PopoverContent`, nor sends the focus to the specific `ref` of the child element set using the `initialFocusRef` prop.

## 💥 Steps to reproduce

1. Open the [CodeSandbox]( https://codesandbox.io/s/fancy-field-muxil) and navigate using the keyboard by pressing the tab key, until `PopoverTrigger` receives the focus.
2. Verify that the `PopoverContent` opens, but the focus stays in the `PopoverTrigger`.
![focus-trigger](https://user-images.githubusercontent.com/6024102/114826496-6d49f100-9d7c-11eb-9b3b-bad042666267.png)

## 💻 Link to reproduction

CodeSandbox reproduction: https://codesandbox.io/s/fancy-field-muxil

## 🧐 Expected behavior

The focus should be sent  to the `PopoverContent` or to the element with the specified `ref` set using the `initialFocusRef`.

Like it does when we use the `Popover` with the `trigger` prop set to `'click'`.
Chakra UI Popover Docs: [Focus on an element when the popover opens](https://chakra-ui.com/docs/overlay/popover#focus-an-element-when-popover-opens)
![focus-element](https://user-images.githubusercontent.com/6024102/114826519-72a73b80-9d7c-11eb-839c-29e5fa1b0c3e.png)

## 🧭 Possible Solution

I believe the problem could be here. We’re only allowing `useFocusOnShow` when the Popover prop `trigger==='click'`:
 https://github.com/chakra-ui/chakra-ui/blob/main/packages/popover/src/use-popover.ts#L189
 

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Chakra UI        |    1.5.1        |
| Browser          |    Chrome 89    |
| Operating System |     Mac OS       |

## 📝 Additional information

-none-

**Comment by segunadebayo** [comment]
Yes, this is by design.

When the popover trigger is set to `hover`, the popover acts as a "tooltip", not a "dialog" hence focus should not move to it. This is why it is crucial to not put important actions in a tooltip-like widget. 

Kindly reconsider this in your product.
